test/e2e/command/botkube.go

```diff
@@ -0,0 +1,151 @@
+package command
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/infracloudio/botkube/pkg/config"
+	"github.com/infracloudio/botkube/pkg/execute"
+	"github.com/infracloudio/botkube/test/e2e/utils"
+	"github.com/nlopes/slack"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type botkubeCommand struct {
+	command  string
+	expected string
+}
+
+// Send botkube command via Slack message and check if BotKube returns correct response
+func (c *context) testBotkubeCommand(t *testing.T) {
+	botkubeVersion := os.Getenv("BOTKUBE_VERSION")
+	// Test cases
+	tests := map[string]botkubeCommand{
+		"BotKube ping": {
+			command:  "ping",
+			expected: fmt.Sprintf("```pong from cluster '%s'\n\nK8s Server Version: %s\nBotKube version: %s```", c.Config.Settings.ClusterName, execute.K8sVersion, botkubeVersion),
+		},
+		"BotKube filters list": {
+			command: "filters list",
+			expected: "FILTER            ENABLED DESCRIPTION\n" +
+				"IngressValidator  true    Checks if services and tls secrets used in ingress specs are available.\n" +
+				"JobStatusChecker  true    Sends notifications only when job succeeds and ignores other job update events.\n" +
+				"NodeEventsChecker true    Sends notifications on node level critical events.\n" +
+				"PodLabelChecker   true    Checks and adds recommedations if labels are missing in the pod specs.\n" +
+				"ImageTagChecker   true    Checks and adds recommendation if 'latest' image tag is used for container image.\n",
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			// Send message to a channel
+			c.SlackServer.SendMessageToBot(c.Config.Communications.Slack.Channel, test.command)
+
+			// Get last seen slack message
+			time.Sleep(time.Second)
+			lastSeenMsg := c.GetLastSeenSlackMessage()
+
+			// Convert text message into Slack message structure
+			m := slack.Message{}
+			err := json.Unmarshal([]byte(lastSeenMsg), &m)
+			assert.NoError(t, err, "message should decode properly")
+			assert.Equal(t, c.Config.Communications.Slack.Channel, m.Channel)
+			switch test.command {
+			case "filters list":
+				fl := compareFilters(strings.Split(test.expected, "\n"), strings.Split(strings.Trim(m.Text, "```"), "\n"))
+				assert.Equal(t, fl, true)
+			default:
+				assert.Equal(t, test.expected, m.Text)
+			}
+		})
+	}
+}
+
+func compareFilters(expected, actual []string) bool {
+	for _, a := range actual {
+		found := false
+		for _, e := range expected {
+			if a == e {
```

Comment:
This checks first line match only and then breaks immediately ?

Reply:
this function is for comparing if two string slices are the same or not. If a string in actual slice match with string in expected slice, we should move to the next string, that's why the break statement.
We should check length of the slices as well here